### DISCUSSION TITLE
(Migration) alternative to `componentsDir`

### DISCUSSION THREAD

**Discussion by JulioJu** [vuepress/core#106] [Q&A] [answered]
Hi,

Thanks for the migration from Vue 2 to Vue 3 and TypeScript :-) 

In the past there was

```
  '@vuepress/register-components',
  {
    componentsDir: [
      path.resolve(__dirname, '../pages'),
      path.resolve(__dirname, 'pages-components'),
    ],
  },
```

But it was removed.

I've checked https://vuepress2.netlify.app/guide/migration.html#for-theme-authors and  https://vuepress2.netlify.app/guide/migration.html#for-theme-authors.  I've redden the doc. 

But I don't see alternative to globally register all files directly under a folder. I see how to import a particular file (`clientAppRootComponentFiles`).

Could you help me please :-) ?

Thanks in advance :-)

**Comment by meteorlxy** [comment, marked as answer]
>  You need to register components manually in `clientAppEnhance.{js,ts}`

I'm not sure if `@vuepress/plugin-register-components` is necessary, as I personally don’t like registering components implicitly 🤔 If there are requests for it, we can provide that plugin. And it's not complicated to write such a plugin yourself

**Reply by JulioJu** [reply]
Thanks a lot,

In fact, the original plugin is very simple https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/plugin-register-components/index.js .

I guess it could be very cool if Vuepress continue to support officially this plugin. 

It was a dependency of Vuepress Core 1.x https://github.com/vuejs/vuepress/blob/38e98634af117f83b6a32c8ff42488d91b66f663/packages/%40vuepress/core/package.json#L31 .

Furthermore, Nuxt.JS has also this feature. 

Should I open an issue ?

**Reply by meteorlxy** [reply]
Of course. We are accepting feature requests.

**Reply by JulioJu** [reply]
Thanks @meteorlxy :-) 

See https://github.com/vuepress/vuepress-next/issues/112
